### PR TITLE
Low: extend support for AWS elastic ip association

### PR DESCRIPTION
**Pull request by guessi**
make it possible to associate secondary private ip address with elastic
ip address, if allocation-id.

if private-ip-address were both set, it will try to associate elastic
interface with the allocation-id, otherwise, it will try to associate
elastic ip with private ip instead (default).

reference docs:
http://docs.aws.amazon.com/cli/latest/reference/ec2/associate-address.html

### CODE REVIEW

**Comment by guessi** [comment]
@krig please review the changes

File: heartbeat/awseip [hunk 1]
```diff
@@ -116,10 +132,24 @@ END
...
-    $AWSCLI ec2 associate-address  \
-        --instance-id ${INSTANCE_ID} \
-        --public-ip ${ELASTIC_IP}
-    RET=$?
+    ASSOCIATE_METHOD="ELASTIC_IP"
+    if [ -n "${ALLOCATION_ID}" ] && [ -n "${PRIVATE_IP_ADDRESS}" ]; then
+        ASSOCIATE_METHOD="ALLOCATION_ID"
+    fi
+
+    if [ "${ASSOCIATE_METHOD}" = "ALLOCATION_ID" ]; then
```
Comment (krig) [inline heartbeat/awseip]: Why the extra condition above, why not just check `[ -n "${ALLOCATION_ID}" ] && [ -n "${PRIVATE_IP_ADDRESS}" ]` here directly?

Comment (guessi) [inline heartbeat/awseip]: Uhh... what a stupid dummy code, I was debugging with this block, I'll remove it later.

**Comment by krig** [comment]
Looks good to me now.

**Comment by krig** [comment]
Squash the two commits together before merging?

**Comment by guessi** [comment]
@krig squashed

**Pull request merged**